src/platform_impl/web/custom_cursor.rs

```diff
@@ -0,0 +1,103 @@
+use crate::cursor::CursorImage;
+use js_sys::{Uint8Array, Uint8ClampedArray};
+use wasm_bindgen::prelude::wasm_bindgen;
+use wasm_bindgen::{JsCast, JsValue};
+use web_sys::{CanvasRenderingContext2d, Document, HtmlCanvasElement, ImageData, Url};
+
+#[derive(Debug, Clone)]
+pub enum WebCustomCursor {
+    Image(CursorImage),
+    Url {
+        url: String,
+        hotspot_x: u32,
+        hotspot_y: u32,
+    },
+}
+
+impl WebCustomCursor {
+    pub fn from_image(image: CursorImage) -> Self {
+        Self::Image(image)
+    }
+}
+
+pub struct CustomCursorInternal {
+    style: String,
+    data_url: Option<String>,
+}
+
+impl CustomCursorInternal {
+    pub fn new(document: &Document, cursor: WebCustomCursor) -> Self {
+        match cursor {
+            WebCustomCursor::Image(image) => Self::from_image(document, image),
+            WebCustomCursor::Url {
+                url,
+                hotspot_x,
+                hotspot_y,
+            } => Self {
+                style: format!("url({}) {} {}, auto", url, hotspot_x, hotspot_y),
+                data_url: None,
+            },
+        }
+    }
+
+    fn from_image(document: &Document, image: CursorImage) -> Self {
+        let cursor_icon_canvas = document
+            .create_element("canvas")
+            .unwrap()
+            .dyn_into::<HtmlCanvasElement>()
+            .unwrap();
+
+        #[allow(clippy::disallowed_methods)]
+        cursor_icon_canvas.set_width(image.width);
+        #[allow(clippy::disallowed_methods)]
+        cursor_icon_canvas.set_height(image.height);
+
+        let context = cursor_icon_canvas
+            .get_context("2d")
+            .unwrap()
+            .unwrap()
+            .dyn_into::<CanvasRenderingContext2d>()
+            .unwrap();
+
+        // Can't create array directly when backed by SharedArrayBuffer.
+        // Adapted from https://github.com/rust-windowing/softbuffer/blob/ab7688e2ed2e2eca51b3c4e1863a5bd7fe85800e/src/web.rs#L196-L223
+        #[wasm_bindgen]
+        extern "C" {
+            #[wasm_bindgen(js_namespace = ImageData)]
+            type ImageDataExt;
+            #[wasm_bindgen(catch, constructor, js_class = ImageData)]
+            fn new(array: Uint8ClampedArray, sw: u32) -> Result<ImageDataExt, JsValue>;
+        }
+        let array = Uint8Array::new_with_length(image.rgba.len() as u32);
+        array.copy_from(&image.rgba);
+        let array = Uint8ClampedArray::new(&array);
+        let image_data = ImageDataExt::new(array, image.width)
+            .map(JsValue::from)
+            .map(ImageData::unchecked_from_js)
```

Comment:
We should also add the `ImageData` crate feature to our `web-sys` dependency.

Reply:
How does it build an work without issues then? But sure if it's needed

Reply:
Probably because some dependency enables it. But if that dependency changes we will be affected, so we don't want to rely on that.